dask/utils.py

```diff
@@ -702,7 +703,8 @@ def infer_storage_options(urlpath, inherit_storage_options=None):
     "host": "node", "port": 123, "path": "/mnt/datasets/test.csv",
     "url_query": "q=1", "extra": "value"}
     """
-    if ':\\' in urlpath:
+    # Handle Windows paths including disk name in this special case
+    if re.match(r'[a-zA-Z]:\\', urlpath):
```

Comment:
I have just noticed that I missed `^` in the regexp, so I amended my PR with this small change just now.